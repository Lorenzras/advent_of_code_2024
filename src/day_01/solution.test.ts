import { describe, expect, it } from "@jest/globals";
import fs from "fs";
import path from "path";
import f from "./solution";

describe("testing add file", () => {
  it("should return 11 according to sample in the problem", () => {
    const input = ["3 4", "4 3", "2 5", "1 3", "3 9", "3 3"].join("\n");

    expect(f(input)).toBe(11);
  });

  it("should return correct answer for a single pair", () => {
    const input = "1 1";

    expect(f(input)).toBe(0);
  });

  it("should return correct answer for the official input data", () => {
    const input = fs.readFileSync(path.join(__dirname, "input.txt"), "utf-8");

    expect(f(input)).toMatchSnapshot();
  });
});
