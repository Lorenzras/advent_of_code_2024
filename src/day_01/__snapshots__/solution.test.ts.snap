// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing add file should return correct answer for the official input data 1`] = `1722302`;
